h5grove/utils.py

```diff
@@ -163,13 +167,18 @@ def convert(data: T, dtype: Optional[str] = "origin") -> T:
 
     if dtype == "safe":
         if not is_numeric_data(data):
-            raise ValueError(f"Unsupported dtype {dtype} for non-numeric content")
+            raise QueryArgumentError(
+                f"Unsupported dtype {dtype} for non-numeric content"
+            )
         return data.astype(_sanitize_dtype(data.dtype), order="C", copy=False)
 
-    raise ValueError(f"Unsupported dtype {dtype}")
+    raise QueryArgumentError(f"Unsupported dtype {dtype}")
+
 
+def is_numeric_data(data: Union[np.ndarray, np.number, np.bool_, bytes]) -> bool:
```

Comment:
Fixed a little oversight in the typing here:  `data` can be `bytes` in case the dataset holds a string value.